pkg/snapshot/snapshot.go

```
@@ -772,6 +802,18 @@ func newOutputConsumer(utxoManager *utxo.Manager) OutputConsumerFunc {
 	}
 }
 
+// returns a treasury output consumer which overrides an existing unspent treasury output with the new one.
+func newUnspentTreasuryOutputConsumer(utxoManager *utxo.Manager) UnspentTreasuryOutputConsumerFunc {
+	return func(output *utxo.TreasuryOutput) error {
+		// delete previous
+		existing, err := utxoManager.UnspentTreasuryOutput()
+		if err == nil {
+			_ = utxoManager.DeleteTreasuryOutput(existing)
```

Maybe add a comment why we can ignore the error here, because the Treasury exists and there shouldn't be more than one unspent entry.

---

Added a comment